proxy/logging/unit-tests/benchmark_LogObject.cc

```diff
@@ -0,0 +1,169 @@
+/** @file
+
+Simple benchmark for LogObject
+
+@section license License
+
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#define CATCH_CONFIG_ENABLE_BENCHMARKING
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#include "LogConfig.h"
+#include "Log.h"
+#include "DiagsConfig.h"
+#include "records/I_RecLocal.h"
+#include "tscore/I_Layout.h"
+
+#include <thread>
+#include <condition_variable>
+#include <chrono>
+
+AppVersionInfo appVersionInfo;
+static char bind_stdout[512] = "";
+static char bind_stderr[512] = "";
+
+namespace notstd
+{
+struct barrier {
+  int count;
+  std::mutex m;
+  std::condition_variable cv;
+
+  barrier(int count) : count(count) {}
+
+  void
+  arrive_and_wait()
+  {
+    std::unique_lock lock{m};
+    if (0 == --count) {
+      cv.notify_all();
+    } else {
+      cv.wait(lock, [this] { return count == 0; });
+    }
+  }
+};
+} // namespace notstd
+
+TEST_CASE("LogObject", "[proxy/logging]")
+{
+  ink_freelist_init_ops(true, true);
+  init_buffer_allocators(0);
+
+  Thread *main_thread = new EThread;
+  main_thread->set_specific();
+
+  auto diagsConfig = new DiagsConfig("Server", "diags.log", "", "", false);
+  // unused, but constructor must be called for side effects.
+  (void)diagsConfig;
+  diags()->set_std_output(StdStream::STDOUT, bind_stdout);
+  diags()->set_std_output(StdStream::STDERR, bind_stderr);
+
+  if (is_debug_tag_set("diags")) {
+    diags()->dump();
+  }
+  Layout::create("/opt/ats");
```

Comment:
The directory doesn't need to be exist, right? It's concerning if it does. No other tests pass a path.

Reply:
This is a  benchmark so you have to go out of your way a bit to run it, but you are correct it shouldn't assume where to put files.  I'll clean this up.  Thanks!

Reply:
After more looking this doesn't look like It changes the filesystem, this just needs to be called so the layout exists.  Also, the files for the benchmark are written to `/tmp`.